src/zmq/zmqpublishnotifier.cpp

```diff
@@ -99,9 +99,8 @@ static bool IsZMQAddressIPV6(const std::string &zmq_address)
     const size_t colon_index = zmq_address.rfind(":");
     if (tcp_index == 0 && colon_index != std::string::npos) {
         const std::string ip = zmq_address.substr(tcp_prefix.length(), colon_index - tcp_prefix.length());
-        CNetAddr addr;
-        LookupHost(ip, addr, false);
-        if (addr.IsIPv6()) return true;
+        const std::optional<CNetAddr> addr{LookupHost(ip, false)};
+        if (addr.has_value() && addr.value().IsIPv6()) return true;
```

Comment:
There was a bug in the old code - if lookup failed it would have returned `true` (i.e. pretend that it is IPv6). The new code fixes this, nice!